### PR TITLE
chore: bump ci dependency versions

### PR DESCRIPTION
**Pull request by holmanb**
## Proposed Commit Message

```
chore: bump ci dependency versions

- use canonical/lxd in integration test
- optimize integration test to run faster
```

## Additional Context
https://github.com/canonical/cloud-init/pull/5658

### CODE REVIEW

**Review by TheRealFalcon — COMMENTED** [review]
I would prefer to keep unit tests on earliest ubuntu, not latest. Us devs are generally all running latest, so I think it helps with additional coverage. We have the 'lowest supported' job, but it winds up being more of an approximation.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -21,69 +21,68 @@ env:
...
         with:
-          # Fetch all tags for tools/read-version
-          fetch-depth: 0
-      - name: Retrieve cloud-init package
-        uses: actions/download-artifact@v3
-        with:
-          name: 'cloud-init-${{ env.RELEASE }}-deb'
+          channel: latest/candidate
+
+      # this is really dumb, but I'm out of time for now
```
[the remark this answers is not in the document]
Comment (TheRealFalcon) [reply .github/workflows/integration.yml:71]: Ah, sorry. I missed that.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -21,69 +21,68 @@ env:
...
           sudo -E su $USER -c 'DEB_BUILD_OPTIONS=nocheck sbuild --nolog --no-run-lintian --no-run-autopkgtest --verbose --dist=${{ env.RELEASE }} --build-dir=${{ runner.temp }} cloud-init_*.dsc'
       - name: Archive debs as artifacts
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: 'cloud-init-${{ env.RELEASE }}-deb'
           path: '${{ runner.temp }}/cloud-init*.deb'
           retention-days: 3
-
-  integration-tests:
```
[the remark this answers is not in the document]
Comment (TheRealFalcon) [reply .github/workflows/integration.yml:57]: The original reasoning is listed here:
https://github.com/canonical/cloud-init/pull/1969#issuecomment-1404630806

cc: @aciba90

**Comment by holmanb** [comment]
> I would prefer to keep unit tests on earliest ubuntu, not latest. Us devs are generally all running latest, so I think it helps with additional coverage. We have the 'lowest supported' job, but it winds up being more of an approximation.

Tests currently run on Jammy which is neither earliest nor latest. Would you rather we drop the release it from Jammy down to Focal? 

My only counterpoint is that as part of Canonical we're better equipped to dogfood new versions so I'd lean towards early adoption since our internal CI infrastructure gets us unittest coverage as part of package build for integration testing.

That said, I really don't think it matters what version of Ubuntu we're on, because for unit tests we use `actions/setup-python`, and install `tox` via `pip`, so I don't see any system dependencies being used anyways.

File: .github/workflows/integration.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheRealFalcon) [reply .github/workflows/integration.yml:57]: I'm ok with it either way, but I'll let @aciba90 chime in first.

**Comment by TheRealFalcon** [comment]
> That said, I really don't think it matters what version of Ubuntu we're on, because for unit tests we use actions/setup-python, and install tox via pip, so I don't see any system dependencies being used anyways.

Good point, +1.

File: .github/workflows/integration.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (aciba90) [reply .github/workflows/integration.yml:57]: I am fine either way too, thanks for fixing this.

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -21,69 +21,65 @@ env:
...
+                --requirements-file test-requirements.txt \
+                --distro ubuntu                           \
+                --system-pkg-names 2> /dev/null           \
+                | tr '\n' ' '                             \
+            )
           sudo sbuild-adduser $USER
           cp /usr/share/doc/sbuild/examples/example.sbuildrc /home/$USER/.sbuildrc
-          # Install all build and test dependencies
-          ./tools/read-dependencies -r requirements.txt -r test-requirements.txt -d ubuntu -s -i
+      - name: Create tmpfs for schroot
```
Comment (TheRealFalcon) [inline .github/workflows/integration.yml:50]: Why is this needed?

File: .github/workflows/integration.yml [hunk 4]
```diff
@@ -21,69 +21,65 @@ env:
 
 jobs:
   package-build:
```
Comment (TheRealFalcon) [inline .github/workflows/integration.yml:23]: We should probably change the name of this now. The name shows up as `Integration tests / package-build`, which made me go looking for a separate "run" or "execute" type job.

**Review by TheRealFalcon — APPROVED** [review]
One question and one ask inline, but +1 either way.

File: .github/workflows/integration.yml (hunk 3 above)
Comment (holmanb) [reply .github/workflows/integration.yml:50]: Eh, it isn't. I was poking around at some optimizations but don't think that this provides much benefit as-is.
